### PR TITLE
DC: Implement the event user-cleanup

### PR DESCRIPTION
**Pull request by stephdl**
https://github.com/NethServer/dev/issues/5624

### CODE REVIEW

File: root/etc/e-smith/events/actions/nethserver-dc-user-delete-cleanup [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+use strict;
+use esmith::event;
+use esmith::ConfigDB;
+
+my $event = shift || die("Missing event argument!");
+my $account = shift || die("Missing account name argument!");
+my $db = esmith::ConfigDB->open_ro() or die("Could not open ConfigDB");
+
+# preserve the data of deleted User
+my $KeepUserData = $db->get_prop('KeepUserData') || 'disabled';
```
Comment (DavidePrincipi) [inline root/etc/e-smith/events/actions/nethserver-dc-user-delete-cleanup]: We are missing a record key here!

**Review by DavidePrincipi — CHANGES_REQUESTED** [review]
We must define (and document) the new key/props. Let's discuss it on community!

**Comment by gsanchietti** [comment]
I don't like much the current approach: we have almost the same code both on nethserver-dc and nethserver-directory.
Can we implement a single action directly into nethserver-sssd? The same can then used also for remote account provider.

Also, I would remove any prop for now and implement it only when automatic removing is in place.
@filippocarletti what do you think?

**Comment by DavidePrincipi** [comment]
> Can we implement a single action directly into nethserver-sssd?

Probably, yes: the data-removal code is the same on both ad and ldap provider

**Comment by stephdl** [comment]
@gsanchietti @DavidePrincipi new way, please consider also https://github.com/NethServer/nethserver-sssd/pull/109

the code to delete the home (user-cleanup) and the code to link user-delete to user-cleanup is now in nethserver-sssd

**Review by DavidePrincipi — APPROVED** [review]

**Pull request merged**